ros_ws/src/car_control/src/car_controller.cpp

```diff
@@ -0,0 +1,67 @@
+#include "car_controller.h"
+
+CarController::CarController()
+    : enabled{ true }
+{
+    this->drive_parameters_subscriber =
+        this->node_handle.subscribe< drive_msgs::drive_param >(
+            TOPIC_DRIVE_PARAM, 1, &CarController::driveParametersCallback,
+            this);
+    this->command_subscriber = this->node_handle.subscribe< std_msgs::String >(
+        TOPIC_COMMAND, 1, &CarController::commandCallback, this);
+
+    this->speed_pulisher =
+        this->node_handle.advertise< std_msgs::Float64 >(TOPIC_FOCBOX_SPEED, 1);
+    this->angle_publisher =
+        this->node_handle.advertise< std_msgs::Float64 >(TOPIC_FOCBOX_ANGLE, 1);
+}
+
+void CarController::driveParametersCallback(
+    const drive_msgs::drive_param::ConstPtr& parameters)
+{
+    this->publishDriveParameters(parameters->velocity, parameters->angle);
+}
+
+void CarController::publishDriveParameters(double raw_speed, double raw_angle)
+{
+    double speed = std::max(0.0, raw_speed * MAX_SPEED);
+    double angle = (raw_angle * MAX_ANGLE + 1) / 2;
+
+    if (this->enabled)
+    {
+        std_msgs::Float64 speed_message;
+        speed_message.data = speed;
```

Comment:
in my experience, it can help to use scoping to emphasise that an object/list/whatever is filled with data.
Here you could write:
```c++
std_msgs::Float64 speed_message;
{
    speed_message.data = speed;
}
```
instead of 
```C++
std_msgs::Float64 speed_message;
speed_message.data = speed;
```